drivers/video/video_stm32_venc.c

```diff
@@ -0,0 +1,1026 @@
+	codingctrl_cfg.inputLineBufDepth = 1;
+	codingctrl_cfg.inputLineBufHwModeEn = 1;
+#else
+	codingctrl_cfg.enableCabac = 0;
+	codingctrl_cfg.quarterPixelMv = 0;
+	codingctrl_cfg.inputLineBufEn = 0;
+	codingctrl_cfg.inputLineBufLoopBackEn = 0;
+	codingctrl_cfg.inputLineBufDepth = 0;
+	codingctrl_cfg.inputLineBufHwModeEn = 0;
+#endif
+	ret = H264EncSetCodingCtrl(data->encoder, &codingctrl_cfg);
+	if (ret != H264ENC_OK) {
+		LOG_ERR("H264EncSetCodingCtrl error=%d\n", ret);
+		return -1;
+	}
+
+	/* set bit rate configuration */
+	ret = H264EncGetRateCtrl(data->encoder, &ratectrl_cfg);
+	if (ret != H264ENC_OK) {
+		LOG_ERR("H264EncGetRateCtrl error=%d\n", ret);
+		return -1;
+	}
+#ifdef VENC_RATECTRL_VBR
+	/* Variable bitrate */
+	ratectrl_cfg.pictureRc = 1;
+	ratectrl_cfg.mbRc = 1;
+	ratectrl_cfg.pictureSkip = 0;
+	ratectrl_cfg.hrd = 0;
+	ratectrl_cfg.qpHdr = VENC_RATECTRL_QP;
+	ratectrl_cfg.qpMin = 10;
+	ratectrl_cfg.qpMax = 51;
+	ratectrl_cfg.gopLen = VENC_RATECTRL_FPS;
+	ratectrl_cfg.bitPerSecond = ((data->width * data->height * 12) * VENC_RATECTRL_FPS) / 30;
+	ratectrl_cfg.intraQpDelta = 0;
+#else
+	/* Constant bitrate */
+	ratectrl_cfg.pictureRc = 0;
+	ratectrl_cfg.mbRc = 0;
+	ratectrl_cfg.pictureSkip = 0;
+	ratectrl_cfg.hrd = 0;
+	ratectrl_cfg.qpHdr = VENC_RATECTRL_QP;
+	ratectrl_cfg.qpMin = VENC_RATECTRL_QP;
+	ratectrl_cfg.qpMax = VENC_RATECTRL_QP;
+#endif
+	ret = H264EncSetRateCtrl(data->encoder, &ratectrl_cfg);
+	if (ret != H264ENC_OK) {
+		LOG_ERR("H264EncSetRateCtrl error=%d\n", ret);
+		return -1;
+	}
+
+	LOG_DBG("%s<\n", __func__);
+	return 0;
+}
+
+static int encoder_start(struct video_stm32_venc_data *data)
+{
+	H264EncRet ret;
+	struct video_buffer *output;
+	H264EncIn encIn = {0};
+	H264EncOut encOut = {0};
+
+	LOG_DBG("%s>\n", __func__);
+
+	output = k_fifo_get(&data->fifo_output_in, K_FOREVER);
+
+	encIn.pOutBuf = (u32 *)output->buffer;
+	encIn.busOutBuf = (uint32_t) encIn.pOutBuf;
+	encIn.outBufSize = ALIGNED_SIZE(output->size);
+
+	/* create stream */
+	ret = H264EncStrmStart(data->encoder, &encIn, &encOut);
+	if (ret != H264ENC_OK) {
+		LOG_ERR("H264EncStrmStart error=%d\n", ret);
+		return -1;
+	}
+
+	output->bytesused = encOut.streamSize;
+	LOG_DBG("SPS/PPS generated, size= %d\n", output->bytesused);
+
+	LOG_DBG("%s> %p\n", __func__, &data->fifo_output_out);
+	k_fifo_put(&data->fifo_output_in, output);
+	k_fifo_put(&data->fifo_output_out, output);
+
+	data->resync = true;
+
+	LOG_DBG("%s<\n", __func__);
+	return 0;
+}
+
+static void  encoder_reset(void)
+{
+	/* VENC HW REset */
+	__HAL_RCC_VENC_FORCE_RESET();
+	k_sleep(K_MSEC(1));
+	__HAL_RCC_VENC_RELEASE_RESET();
+	k_sleep(K_MSEC(1));
+}
+
+static int encode_frame(struct video_stm32_venc_data *data)
+{
+	int ret = H264ENC_FRAME_READY;
+	struct video_buffer *input;
+	struct video_buffer *output;
+	H264EncIn encIn = {0};
+	H264EncOut encOut = {0};
+
+	LOG_DBG("%s>\n", __func__);
+
+	if (!data->encoder) {
+		ret = encoder_prepare(data);
+		if (ret)
+			return -1;
```

Comment:
```suggestion
		if (ret) {
			return -1;
		}
```
or even `ret < 0` and propagate `ret`

Reply:
This is not same coding style than Linux kernel, is there a doc pointing out the coding style to match ? Or better: a script like Linux kernel "checkpatch" ?